scalalib/src/mill/scalalib/PublishModule.scala

```diff
@@ -73,32 +73,35 @@ trait PublishModule extends JavaModule { outer =>
   }
 
   def publish(sonatypeCreds: String,
-              gpgPassphrase: String,
+              gpgPassphrase: String = "",
+              signed: Boolean = true,
               release: Boolean): define.Command[Unit] = T.command {
     val PublishModule.PublishData(artifactInfo, artifacts) = publishArtifacts()
     new SonatypePublisher(
       sonatypeUri,
       sonatypeSnapshotUri,
       sonatypeCreds,
-      gpgPassphrase,
+      Some(gpgPassphrase).filter(_.nonEmpty),
```

Comment:
If having `gpgPassphrase` parameter optional is not possible, I guess we could have default value as `null` and wrap it in `Option(gpgPassphrase)` instead of filtering out an empty string, what do you think?

Reply:
Addressed ec362d8